### PR TITLE
Entity

### PR DESCRIPTION
**Pull request by srikanthgudipudi**
Entity module

### CODE REVIEW

File: src/app/modules/admin-core/components/bread-crum/bread-crum.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div *ngIf="edit != 'edit/:id'" class="pageName">
+<div *ngIf ="show" class="pageName">
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/components/bread-crum/bread-crum.component.html:1]: Why is this condition??

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -23,7 +25,12 @@ const routes: Routes = [
     canActivate: [UsersGuard],
     loadChildren: './modules/upload-status/upload-csv.module#UploadCSVModule'
   },
-  { path: 'home', component: HomeComponent, data: { title: [{name: 'Admin Console', link: '/home'}]} },
+  {
+    path: 'entities',
+    canActivate: [UsersGuard],
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts]: this should be enabled only for platform admins

File: src/app/modules/entities/entities-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { ViewEntityDetailsComponent } from './view-entity-details/view-entity-details.component';
+
+const routes: Routes = [
+  {
+    path: '',
+    // data: { title: 'Home' },
+    children: [
+      {
+        path: 'list', component: StateEntityListComponent, data: {
+          title: [{ name: 'Home >', link: '/home' },
```
Comment (kiranharidas187) [inline src/app/modules/entities/entities-routing.module.ts]: we discussed to remove the arrow and put it inside the breadcrumb component

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        </mat-button-toggle>
+        <mat-button-toggle matTooltip="Inactive organisations" (click)="allOrganisation(toggleGroup.value)"
+          class="filter" value="Inactive">Inactive
+        </mat-button-toggle>
+      </mat-button-toggle-group> -->
+    </div>
+
+    <div class="row" style="padding-right: 3%;">
+      <!-- <div class="btn-gap">
+        <button mat-button (click)="addNewOrganisation()" matTooltip="Add Organisation">
```
Comment (kiranharidas187) [inline src/app/modules/entities/state-entity-list/state-entity-list.component.html]: remove unwanted code

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          </span>
+        </ng-container>
+      </span>
+      <mat-header-row *matHeaderRowDef="displayedColumns;sticky: true" class="p-button">
+      </mat-header-row>
+      <mat-row *matRowDef="let row; columns: displayedColumns;">
+      </mat-row>
+    </mat-table>
+
+    <div class="message" *ngIf="noData || !dataSource">No records found</div>
```
Comment (kiranharidas187) [inline src/app/modules/entities/state-entity-list/state-entity-list.component.html]: Please make a component and reuse it

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    <mat-paginator [length]="recordCount" [showFirstLastButtons]="true" [pageSize]="10"
+      [pageSizeOptions]="[5, 10, 25, 100]">
```
Comment (kiranharidas187) [inline src/app/modules/entities/state-entity-list/state-entity-list.component.html]: Add this as a constant in the core module and reuse

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  constructor(private entityService: EntityService,
+    private commonServiceService: CommonServiceService,
+    private router: Router) { }
+
+  ngOnInit() {
+    this.getEntityStateList();
+  }
+
+  applyFilter(event: Event) {
+    const filterValue = (event.target as HTMLInputElement).value;
```
Comment (kiranharidas187) [inline src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: Why is this HTMLInputElement

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+  // get color based on the status
+  getItemCssClassByStatus(status): string {
+    switch (status) {
+      case 'Active':
+        return 'active';
+      case 'Inactive':
+        return 'inactive';
+    }
+    return '';
```
Comment (kiranharidas187) [inline src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: why we need this return

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+            <div>
+              <!-- <mat-icon class="c-mat-icon">account_circle</mat-icon> -->
+              <div class="text-heading">
+                State Hierarchy
+              </div>
+
+            </div>
+          </mat-card-content>
+          <div class=" row c-line"></div>
+          <div style="display: flex;flex-direction: column">
```
Comment (kiranharidas187) [inline src/app/modules/entities/view-entity-details/view-entity-details.component.html]: use class

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+            <span style="padding:10px;" *ngFor="let data of childHierarchary">
+              <button [style.background-color]="data === type ? '#e0e0e0': '#f5f5f5'" style="width:100%; height: 50px;"
```
Comment (kiranharidas187) [inline src/app/modules/entities/view-entity-details/view-entity-details.component.html]: use class

File: src/app/modules/entities/view-entity-details/view-entity-details.component.ts [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+
+  getItemCssClassByStatus(status): string {
+    switch (status) {
+      case 'Active':
+        return 'active';
+      case 'Inactive':
+        return 'inactive';
+    }
+    return '';
```
Comment (kiranharidas187) [inline src/app/modules/entities/view-entity-details/view-entity-details.component.ts]: why this return

File: src/app/modules/admin-core/components/bread-crum/bread-crum.component.html (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/components/bread-crum/bread-crum.component.html:1]: In order to show the editing record in breadscrum

File: src/app/app-routing.module.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts]: ok , changed

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.html]: Removed

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: we are returning the class when other than this we are passing empty

File: src/app/modules/entities/view-entity-details/view-entity-details.component.ts (hunk 11 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/view-entity-details/view-entity-details.component.ts]: Removed

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html (hunk 10 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/view-entity-details/view-entity-details.component.html]: replaced with class

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html (hunk 9 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/view-entity-details/view-entity-details.component.html]: Replaced

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: To get the value of the input field, catching through event

File: src/app/modules/entities/entities-routing.module.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/entities-routing.module.ts]: Changed

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html (hunk 5 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.html]: changed

File: src/app/modules/admin-shared/no-record/no-record.component.html [hunk 12]
```diff
@@ -0,0 +1 @@
+<div class="message" *ngIf="recordCount == 0 || !dataSource">No records found</div>
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/no-record/no-record.component.html]: this condition should be only in corresponding pages

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 7 above)
Comment (kiranharidas187) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: We don't need HTMLInputElement to get the value of field please check

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+              </span>
+              <mat-header-row *matHeaderRowDef="displayedColumns;sticky: true" class="p-button">
+              </mat-header-row>
+              <mat-row *matRowDef="let row; columns: displayedColumns;">
+              </mat-row>
+            </mat-table>
+
+            <div class="message" *ngIf="recordCount == 0 || !dataSource">No records found</div>
+            <mat-paginator [length]="recordCount" [showFirstLastButtons]="true" [pageSize]="10"
+              [pageSizeOptions]="[5, 10, 25, 100]">
```
Comment (kiranharidas187) [inline src/app/modules/entities/view-entity-details/view-entity-details.component.html]: use constants to add here

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 8 above)
Comment (kiranharidas187) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: there is a default keyword in switch case,,. Please use that

File: src/app/modules/admin-core/components/bread-crum/bread-crum.component.html [hunk 14]
```diff
@@ -1,5 +1,6 @@
-<div *ngIf="edit != 'edit/:id'" class="pageName">
-    <span *ngFor="let data of crumData?.title">
+<div *ngIf ="show" class="pageName">
+    <span *ngFor="let data of crumData?.title; let i = index">
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/components/bread-crum/bread-crum.component.html]: search for last in ngFor

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: Removed that empty also

File: src/app/modules/admin-shared/no-record/no-record.component.html (hunk 12 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/no-record/no-record.component.html]: yes removed

File: src/app/modules/entities/state-entity-list/state-entity-list.component.ts (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.ts]: Removed

File: src/app/modules/entities/state-entity-list/state-entity-list.component.html (hunk 6 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/state-entity-list/state-entity-list.component.html:67]: Can you please explain this bro

File: src/app/modules/entities/view-entity-details/view-entity-details.component.html (hunk 13 above)
Comment (srikanthgudipudi) [reply src/app/modules/entities/view-entity-details/view-entity-details.component.html]: can you please explain to get clear idea?

File: src/app/modules/admin-core/components/bread-crum/bread-crum.component.html (hunk 14 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/components/bread-crum/bread-crum.component.html]: yes i changed to last, but is it makes any difference ?

**Pull request merged**